### ISSUE TITLE
Borzoi generate too slow for big projects

### ISSUE THREAD

**Issue by theishanbh** [automation-co/borzoi#18]
One thing I noticed is that when I ran the `borzoi generate` command for bigger projects that had a lot of dependencies , such as react projects or other heavy NodeJS based projects. That is checking through all the files. In my case, I think the folder `node_modules` is too big, which is slowing down the implementation. 

For this, I suggest that we use `return filepath.SkipDir` which can skip going through a subdirectory. 

```
filepath.WalkDir(path, func(path string, info fs.DirEntry, err error) error {
  if err != nil {
	  log.Fatalf(err.Error())
  }
  // checks a subdirectory before .git folder if a folder contains .git folder
  if info.IsDir() {
	  if _, err := os.Stat(path + "\\.git"); !os.IsNotExist(err) {
		  return filepath.SkipDir
         }
   }
}
```
Basically instead of checking all the files and folders for git repo, we only instead check upto only the sub-directory before the direcotry which has `.git` folder to make the command avoid the hassle of checking through all the dependencies of the project as it useless and adds extra time.

**Comment by Aarush-Goyal** [comment]
This is a good suggestion.
Combined with concurrency issues #5 the project will become a lot faster.
